File: problems/21/solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def mergeTwoLists(self, l1: ListNode, l2: ListNode) -> ListNode:
        node = ListNode(0)
        tracker = node
        
        while l1 and l2:
            val = None
            
            if l1.val <= l2.val:
                val = l1.val
                l1 = l1.next
            else:
                val = l2.val
                l2 = l2.next
            
            tracker.next = ListNode(val)
            tracker = tracker.next
        
        tracker.next = l1 if l1 else l2
        
        return node.next

File: problems/21/solution.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} l1
 * @param {ListNode} l2
 * @return {ListNode}
 */
var mergeTwoLists = function(l1, l2) {
    const node = new ListNode(0);
    let tracker = node;
    
    while (l1 !== null && l2 !== null) {
        let val;
        
        if (l1.val <= l2.val) {
            val = l1.val;
            l1 = l1.next;
        } else {
            val = l2.val;
            l2 = l2.next;
        }
        
        tracker.next = new ListNode(val);
        tracker = tracker.next;
    }
    
    tracker.next = l1 !== null ? l1 : l2;
    
    return node.next;
};